cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -262,7 +318,8 @@ public void update(Id.DatasetInstance instance, Map<String, String> properties)
    * @throws DatasetNotFoundException if the dataset instance was not found
    * @throws IOException if there was a problem in checking if the namespace exists over HTTP
    */
-  public void drop(Id.DatasetInstance instance) throws Exception {
+  void drop(Id.DatasetInstance instance) throws Exception {
+    authorizationEnforcementService.enforce(instance.toEntityId(), SecurityRequestContext.toPrincipal(), Action.ADMIN);
```

Comment:
yes, the pattern so far has been: to create an entity, you need write on its parent. Is that a fair policy?